fixbackend/domain_events/consumers.py

```diff
@@ -78,3 +82,51 @@ def __init__(
     async def process_user_registered_event(self, event: UserRegistered) -> None:
         message = Signup(recipient=event.email)
         await self.notification_service.send_message(to=event.email, message=message)
+
+
+class ScheduleTrialEndReminder(Service):
+    def __init__(
+        self,
+        subscriber: DomainEventSubscriber,
+        one_time_email_service: OneTimeEmailService,
+    ) -> None:
+        self.one_time_email_service = one_time_email_service
+        # uncomment once we have payment
+        # subscriber.subscribe(WorkspaceCreated, self.schedule_trial_end_reminder, "schedule_trial_end_reminder")
+
+    async def schedule_trial_end_reminder(self, event: WorkspaceCreated) -> None:
+        for days_before_expiration in [2, 1]:
+            await self.one_time_email_service.schedule_one_time_email(
+                kind=OneTimeEmailKind.TrialEndNotification,
+                scheduled_at=utc() + trial_period_duration() - timedelta(days=days_before_expiration),
```

Comment:
Mmmh that's something we can not merge atm, or?
The trial period is now() + 14d atm and is constantly changing. I don't want to send an email in 14 days.
Also: this code will not work for all existing workspaces.